File: app/src/main/java/com/noahseidman/digiid/utils/SeedUtil.java
package com.noahseidman.digiid.utils;

import com.noahseidman.digiid.MainActivity;

import javax.annotation.Nonnull;
import java.security.SecureRandom;
import java.util.List;
import java.util.Locale;

public class SeedUtil {

    @Nonnull
    public static String generateRandomSeed(@Nonnull final MainActivity ctx) {
        SecureRandom sr = new SecureRandom();
        final String[] words;
        List<String> list;
        String languageCode = Locale.getDefault().getLanguage();
        if (languageCode == null) {
            languageCode = "en";
        }
        list = Bip39Reader.bip39List(ctx, languageCode);
        words = list.toArray(new String[0]);
        final byte[] randomSeed = sr.generateSeed(32);
        return new String(ctx.encodeSeed(randomSeed, words));
    }
}

File: app/src/main/java/com/noahseidman/digiid/utils/SeedUtil.kt
package com.noahseidman.digiid.utils

import android.content.Context
import java.security.SecureRandom
import java.util.*

object SeedUtil {

    external fun encodeSeed(seed: ByteArray, wordList: Array<String>): ByteArray

    fun generateRandomSeed(ctx: Context): String {
        var languageCode: String? = Locale.getDefault().language
        if (languageCode == null) {
            languageCode = "en"
        }
        val list = Bip39Reader.bip39List(ctx, languageCode)
        val randomSeed = SecureRandom().generateSeed(32)
        return String(encodeSeed(randomSeed, list.toTypedArray()))
    }

    fun getWordList(ctx: Context): List<String> {
        var languageCode: String? = Locale.getDefault().language
        if (languageCode == null) {
            languageCode = "en"
        }
        return Bip39Reader.bip39List(ctx, languageCode)
    }
}